src/BinaryInstaller.php

```diff
@@ -196,10 +196,15 @@ protected function installBinary($binary, $version, $url, $sha, $hashalgo = 'sha
         $httpDownloader = Factory::createHttpDownloader($this->io, $this->config);
         $parts = explode('/', $url);
         $fileName = array_pop($parts);
-        $cacheFolder = $this->cache->getRoot().$binary.\DIRECTORY_SEPARATOR.$version;
-        $cacheDestination = $cacheFolder.\DIRECTORY_SEPARATOR.$fileName;
-        $cacheExtractedBinary = $cacheFolder.\DIRECTORY_SEPARATOR.$binary;
-        $binDestination = $bin.\DIRECTORY_SEPARATOR.$binary;
+        if ($this->cache->isEnabled()) {
+            $cacheFolder = $this->cache->getRoot() . $binary . \DIRECTORY_SEPARATOR . $version;
+        }
+        else {
+            $cacheFolder =  sys_get_temp_dir() . \DIRECTORY_SEPARATOR . $binary . \DIRECTORY_SEPARATOR . $version;
```

Comment:
```suggestion
            $cacheFolder = sys_get_temp_dir() . \DIRECTORY_SEPARATOR . $binary . \DIRECTORY_SEPARATOR . $version;
```

Reply:
I actually wonder if in the `--no-cache` case, we should... not cache? This looks to be caching in the system temp directory, which means we're more behaving like "don't cache in the usual composer dir".

Reply:
@deviantintegral it won't retrieve it from the tmp directory on the next install and will do a fresh download, it's just being stashed there before being unzipped